pyqtorch/hamiltonians/evolution.py

```diff
@@ -315,6 +322,55 @@ def spectral_gap(self) -> Tensor:
         spectral_gap = torch.unique(torch.abs(torch.tril(diffs)))
         return spectral_gap[spectral_gap.nonzero()]
 
+    def _forward(
+        self,
+        state: Tensor,
+        values: dict[str, Tensor] | ParameterDict = dict(),
+        embedding: Embedding | None = None,
+    ) -> State:
+        evolved_op = self.tensor(values, embedding)
+        return apply_operator(
+            state=state, operator=evolved_op, qubit_support=self.qubit_support
+        )
+
+    def _forward_time(
```

Comment:
I assume it wasn't straightforward to merge it with the `_forward` above ?

Reply:
well, this refactoring is the direct consequence of @jpmoutinho comment from above that suggested separating both cases (time-dependent and independent) into different functions. Before this refactoring everything was in `forward()`

Reply:
OK no worries. I was just curious about the design.